src/main.rs

```diff
@@ -80,6 +82,58 @@ async fn inner_main() -> Result<i32> {
     let mut client = Client::builder().redirect(redirect);
     let mut resume: Option<u64> = None;
 
+    if url.0.scheme() == "https" {
+        client = client.danger_accept_invalid_certs(args.verify == Verify::No);
+
+        if let Verify::CustomCABundle(path) = args.verify {
+            // FIXME: Change the version of reqwest in Cargo.toml to an official released version (> 0.11.0) which should contain this function
+            // See: https://github.com/seanmonstar/reqwest/pull/1150
+            client = client.tls_built_in_root_certs(false);
+
+            let mut buffer = Vec::new();
+            let mut file = File::open(&path).with_context(|| {
+                format!("Failed to open the custom CA bundle: {}", path.display())
+            })?;
+            file.read_to_end(&mut buffer).with_context(|| {
+                format!(
+                    "Failed to read the custom CA bundle file: {}",
+                    path.display()
+                )
+            })?;
+
+            for pem in pem::parse_many(buffer) {
+                let certificate = reqwest::Certificate::from_pem(pem::encode(&pem).as_bytes())
+                    .with_context(|| {
+                        format!("Failed to load the custom CA bundle: {}", path.display())
+                    })?;
+                client = client.add_root_certificate(certificate);
+            }
+        };
+
+        if let Some(cert) = args.cert {
+            let mut buffer = Vec::new();
+            let mut file = File::open(&cert)
+                .with_context(|| format!("Failed to open the cert file: {}", cert.display()))?;
+            file.read_to_end(&mut buffer)
+                .with_context(|| format!("Failed to read the cert file: {}", cert.display()))?;
+
+            if let Some(cert_key) = args.cert_key {
+                buffer.push(0x0Au8);
+
+                let mut file = File::open(&cert_key).with_context(|| {
+                    format!("Failed to open the cert key file: {}", cert_key.display())
+                })?;
+                file.read_to_end(&mut buffer).with_context(|| {
+                    format!("Failed to read the cert key file: {}", cert_key.display())
+                })?;
+            }
+
+            let identity = reqwest::Identity::from_pem(&buffer)
```

Comment:
This binds us to `rustls`. That's ok, I think, but it's relevant to #49.

Reply:
Should we document it on somewhere?

Reply:
Just having this mention show up in that issue is enough, I think. We already had `rustls` unconditionally enabled, so it's not a user-visible change.

Reply:
I hadn't noticed this function being behind a feature flag. And now that you mention it: the other `Identity` constructor is behind another feature, which is for `native-tls`. This is a little surprising, as that means there is no way to use client certificates with reqwest in a tls implementation agnostic way.